### ISSUE TITLE
In-Game: Silent crash caused by built-in audio system

### ISSUE THREAD

**Issue by KeeVeeGames** [YoYoGames/GameMaker-Bugs#5942]
### Description
The game crashes after several minutes of playing and moving between rooms with the classic silent crash message:
```
Runner.exe exited with non-zero status (-1073741819)
```

Debugging with VS revealed the details of the crash:

```
0xC0000005: Access violation reading location 0x0000000000000000.

AudioMixer::Process(float *,int,int)
aluMixData(struct ALCdevice_struct *,void *,int,bool)
ALCdevice_wasapi::ThreadFunc(void *)
CThread::WaitForExit(void)
kernel32.dll!00007fffa78d7344()
ntdll.dll!00007fffa7fc26b1()
```

I am using my own partially implemented wrapper around audio system presented as `Audio` "static-classy globalvar", sorry for possible confusion. It uses only standard audio functions inside, no additional libraries and such. I couldn't narrow down the exact audio functionality causing the crash, but it seems something with audio buses possibly, because commenting out the code related to `audioBusLevel` variable seem to fix the issue or at least prolong the crash for more than 90 minutes.

Possible relations:
https://github.com/YoYoGames/GameMaker-Bugs/issues/5937
https://github.com/YoYoGames/GameMaker-Bugs/issues/5643
https://forum.gamemaker.io/index.php?threads/high-crash-rate-in-v-2023-8-1-102.106111/

The crash is reproducable on both VM and YYC, on Windows and Android targets.

### Steps To Reproduce
Couldn't reproduce that in the small project, so had to send the full game instead. I added an "autoplay" that just jump into the level and then go back to the menu, because it also seem that the bug is related to room switches.

1. Run the game
2. Press the "Library" menu option
3. Autoplay will start to move between the level and Library menu.
4. Wait for 2-40 mins for crash to happen.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.4.1.152 Runtime v2024.4.1.202

### Which operating system(s) are you seeing the problem on?
Windows 10.0.19045.0

### Which platform(s) are you seeing the problem on?
Android, Windows

[54bfb4ef-b6a2-41ae-b510-325db6e126f3](https://api.gamemaker.io/api/github/downloads/54bfb4ef-b6a2-41ae-b510-325db6e126f3)

**Comment by toby-yoyo** [comment]
Fixed a thread synchronisation issue relating to audio bus destruction.

**Comment by KeeVeeGames** [comment]
> Fixed a thread synchronisation issue relating to audio bus destruction.

Not to argue, just to clarify, the project attached uses one bus that is never destroyed.

**Comment by toby-yoyo** [comment]
It looks like there's a call to `audio_bus_create` in your player object, which is called each time you transition from the library to the level.

**Comment by KeeVeeGames** [comment]
You are absolutely right, thank you!

**Comment by Emc1923** [comment]
Verified fixed in  IDE v2024.600.0.567  Runtime v2024.600.0.588.
